services/provider/server/consumer/store.go

```diff
@@ -0,0 +1,68 @@
+package consumer
+
+import (
+	"errors"
+	"sync"
+)
+
+// Consumer is the storage consumer
+type Consumer struct {
+	Name  string
+	ID    string
+	Token string
+}
```

Comment:
Is there a reason why we use a local struct here instead of the struct representing the StorageConsumer CR?

Reply:
Just a in-memory store (cache) to track all the consumers and tokens used by them.